Non-Composed nodes have a keepout and speed zone substitution missing

We added https://github.com/ros-navigation/navigation2/blob/main/nav2_bringup/launch/bringup_launch.py#L54-L57 to bringup launch file which is used by the component container and thereby component nodes. 

When using a non-component node launch, that isn't being set https://github.com/ros-navigation/navigation2/blob/main/nav2_bringup/launch/navigation_launch.py#L59-L69 yielding and error:

>   [controller_server-10] [ERROR] [1752050492.827177333] [robot1.local_costmap.local_costmap]: Parameter problem: parameter 'keepout_filter.enabled' has invalid type: Wrong parameter type, parameter {keepout_filter.enabled} is of type {bool}, setting it to {string} is not allowed.

--- a/nav2_bringup/launch/bringup_launch.py
+++ b/nav2_bringup/launch/bringup_launch.py
@@ -235,6 +235,8 @@ def generate_launch_description() -> LaunchDescription:
                     'params_file': params_file,
                     'use_composition': use_composition,
                     'use_respawn': use_respawn,
+                    'use_keepout_zones': use_keepout_zones,
+                    'use_speed_zones': use_speed_zones,
                     'container_name': 'nav2_container',
                 }.items(),
             ),

--- a/nav2_bringup/launch/navigation_launch.py
+++ b/nav2_bringup/launch/navigation_launch.py
@@ -38,6 +38,8 @@ def generate_launch_description() -> LaunchDescription:
     container_name_full = (namespace, '/', container_name)
     use_respawn = LaunchConfigAsBool('use_respawn')
     log_level = LaunchConfiguration('log_level')
+    use_keepout_zones = LaunchConfigAsBool('use_keepout_zones')
+    use_speed_zones = LaunchConfigAsBool('use_speed_zones')
 
     lifecycle_nodes = [
         'controller_server',
@@ -58,11 +60,17 @@ def generate_launch_description() -> LaunchDescription:
     # Create our own temporary YAML files that include substitutions
     param_substitutions = {'autostart': autostart}
 
+    yaml_substitutions = {
+        'KEEPOUT_ZONE_ENABLED': use_keepout_zones,
+        'SPEED_ZONE_ENABLED': use_speed_zones,
+    }
+
     configured_params = ParameterFile(
         RewrittenYaml(
             source_file=params_file,
             root_key=namespace,
             param_rewrites=param_substitutions,
+            value_rewrites=yaml_substitutions,
             convert_types=True,
         ),
         allow_substs=True,
@@ -119,6 +127,16 @@ def generate_launch_description() -> LaunchDescription:
 
     declare_log_level_cmd = DeclareLaunchArgument(
         'log_level', default_value='info', description='log level'
+    )
+
+    declare_use_keepout_zones_cmd = DeclareLaunchArgument(
+        'use_keepout_zones', default_value='True',
+        description='Whether to enable keepout zones or not'
+    )
+
+    declare_use_speed_zones_cmd = DeclareLaunchArgument(
+        'use_speed_zones', default_value='True',
+        description='Whether to enable speed zones or not'
     )
 
     load_nodes = GroupAction(
@@ -351,6 +369,8 @@ def generate_launch_description() -> LaunchDescription:
     ld.add_action(declare_container_name_cmd)
     ld.add_action(declare_use_respawn_cmd)
     ld.add_action(declare_log_level_cmd)
+    ld.add_action(declare_use_keepout_zones_cmd)
+    ld.add_action(declare_use_speed_zones_cmd)
     # Add the actions to launch all of the navigation nodes
     ld.add_action(load_nodes)
     ld.add_action(load_composable_nodes)

